src/hivpy/population.py

```diff
@@ -78,20 +143,23 @@ def transform_group(self, param_list, func, use_size=True, sub_pop=None):
         `sub_pop` is `None` by default, in which case the transform acts upon the entire dataframe.
         If `sub_pop` is defined, then it acts only on the part of the dataframe defined
         by `data.loc[sub_pop]`"""
-        # HIV_STATUS is just a dummy column to allow us to use the transform method
+        # Use Dummy column to in order to enable transform method and avoid any risks to data
         def general_func(g):
             if len(param_list) == 1:
                 args = [g.name]
             else:
                 args = list(g.name)
+
+            g = list(map(lambda x: self.get_correct_column(x), g))
+
```

Comment:
Not sure what this is doing, but it looks like it will not work with the following lines (`g` will be a list and will not longer have a `size` attribute).

Should the second argument to `map` be `args` instead of `g`?

Reply:
Also, you don't need the lambda here. You could simply `map(self.get_correct_column, ...)`

Reply:
This was mostly just wrong and has been moved somewhere else; I think I was tired when I wrote this! This map applies to the list of params, to get the columns for the correct time periods.